upstairs/src/test.rs

```diff
@@ -3831,268 +3832,306 @@ mod up_test {
-        assert!(guest.consume_req().is_none());
+        assert!(guest.consume_req().await.is_none());
 
         // Don't use guest.read, that will send a block size query that will
         // never be answered.
-        let _ = guest.send(BlockOp::Read {
-            offset: Block::new_512(0),
-            data: Buffer::new(1),
-        });
-        let _ = guest.send(BlockOp::Read {
-            offset: Block::new_512(0),
-            data: Buffer::new(8000),
-        });
-        let _ = guest.send(BlockOp::Read {
-            offset: Block::new_512(0),
-            data: Buffer::new(16000),
-        });
+        let _ = guest
+            .send(BlockOp::Read {
+                offset: Block::new_512(0),
+                data: Buffer::new(1),
+            })
+            .await;
+        let _ = guest
+            .send(BlockOp::Read {
+                offset: Block::new_512(0),
+                data: Buffer::new(8000),
+            })
+            .await;
+        let _ = guest
+            .send(BlockOp::Read {
+                offset: Block::new_512(0),
+                data: Buffer::new(16000),
+            })
+            .await;
 
         // First two reads succeed
-        assert!(guest.consume_req().is_some());
-        assert!(guest.consume_req().is_some());
+        assert!(guest.consume_req().await.is_some());
+        assert!(guest.consume_req().await.is_some());
 
         // Next cannot be consumed until there's available IOP tokens so it
         // remains in the queue.
-        assert!(guest.consume_req().is_none());
-        assert!(!guest.reqs.lock().unwrap().is_empty());
-        assert_eq!(*guest.iop_tokens.lock().unwrap(), 2);
+        assert!(guest.consume_req().await.is_none());
+        assert!(!guest.reqs.lock().await.is_empty());
+        assert_eq!(*guest.iop_tokens.lock().await, 2);
 
         // Replenish one token, meaning next read can be consumed
-        guest.leak_iop_tokens(1);
-        assert_eq!(*guest.iop_tokens.lock().unwrap(), 1);
+        guest.leak_iop_tokens(1).await;
+        assert_eq!(*guest.iop_tokens.lock().await, 1);
 
-        assert!(guest.consume_req().is_some());
-        assert!(guest.reqs.lock().unwrap().is_empty());
-        assert_eq!(*guest.iop_tokens.lock().unwrap(), 2);
+        assert!(guest.consume_req().await.is_some());
+        assert!(guest.reqs.lock().await.is_empty());
+        assert_eq!(*guest.iop_tokens.lock().await, 2);
 
-        guest.leak_iop_tokens(2);
-        assert_eq!(*guest.iop_tokens.lock().unwrap(), 0);
+        guest.leak_iop_tokens(2).await;
+        assert_eq!(*guest.iop_tokens.lock().await, 0);
 
-        guest.leak_iop_tokens(16000);
-        assert_eq!(*guest.iop_tokens.lock().unwrap(), 0);
+        guest.leak_iop_tokens(16000).await;
+        assert_eq!(*guest.iop_tokens.lock().await, 0);
 
         Ok(())
     }
 
-    #[test]
-    fn test_flush_does_not_consume_iops() -> Result<()> {
+    #[tokio::test(flavor = "multi_thread", worker_threads = 8)]
```

Comment:
Why did you add the worker threads here, but not elsewhere?

Reply:
That was a mistake, corrected in ec2207f